modules/core/src/main/java/org/apache/ignite/internal/processors/closure/GridClosureProcessor.java

```diff
@@ -960,9 +962,11 @@ public void runLocalWithThreadPolicy(IgniteThread thread, Runnable c) {
 
             final GridWorkerFuture<R> fut = new GridWorkerFuture<>();
 
+            final SecurityContext secCtx = ctx.security().securityContext();
```

Comment:
An operation can be performed without a security context if the worker was rejected.

Reply:
No, it will be executed in current context, which we give to worker. So, in case of reject we already have proper context.

Reply:
Got it, thanks.